src/core/thread/mle.cpp

```diff
@@ -4357,8 +4357,18 @@ void Mle::LogError(MessageAction aAction, MessageType aType, Error aError)
 {
     if (aError != kErrorNone)
     {
-        otLogWarnMle("Failed to %s %s%s: %s", aAction == kMessageSend ? "send" : "process", MessageTypeToString(aType),
-                     MessageTypeActionToSuffixString(aType, aAction), ErrorToString(aError));
+        if (aAction == kMessageReceive && (aError == kErrorDrop || aError == kErrorNoRoute))
+        {
+            otLogInfoMle("Failed to %s %s%s: %s", aAction == kMessageSend ? "send" : "process",
```

Comment:
Given the if condition `aAction == kMessageReceive`, aAction should not be kMessageSend for sure.

```suggestion
            otLogInfoMle("Failed to process %s%s: %s", 
```

Reply:
This change will increase code size since we are introducing new string constants. 

Maybe better use `otLogInfoMle("Failed to %s %s%s: %s", "process",`